hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/TestPrefetch.java

```diff
@@ -364,16 +366,20 @@ public void testPrefetchWithDelay() throws Exception {
     // Wait for 20 seconds, no thread should start prefetch
     Thread.sleep(20000);
     assertFalse("Prefetch threads should not be running at this point", reader.prefetchStarted());
-    while (!reader.prefetchStarted()) {
-      assertTrue("Prefetch delay has not been expired yet",
-        getElapsedTime(startTime) < PrefetchExecutor.getPrefetchDelay());
-    }
-    if (reader.prefetchStarted()) {
-      // Added some delay as we have started the timer a bit late.
+    long timeout = 10000;
+    while (!reader.prefetchStarted() && !reader.prefetchComplete()) {
+      // Wait until the prefetch is triggered.
       Thread.sleep(500);
-      assertTrue("Prefetch should start post configured delay",
-        getElapsedTime(startTime) > PrefetchExecutor.getPrefetchDelay());
+      if (timeout <= 0) break;
+      timeout -= 500;
     }
+    assertTrue(reader.prefetchStarted() || reader.prefetchComplete());
+
+    // Added some delay as we have started the timer a bit late.
+    Thread.sleep(500);
```

Comment:
Why do we need this extra sleep after the loop above?

Reply:
This was present earlier along with the comment at line 378.

Reply:
I will try to remove this one and check if the test passes.

Reply:
ack